python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py

```diff
@@ -35,42 +36,72 @@ def _get_node_name(node_info: Mapping[str, Any]):
     return "__".join([node_info["resource_type"], node_info["package_name"], node_info["name"]])
 
 
-def _dbt_node_to_asset(
-    node_info: Mapping[str, Any],
+def _dbt_nodes_to_assets(
+    node_infos: Sequence[Mapping[str, Any]],
     runtime_metadata_fn: Optional[
         Callable[[SolidExecutionContext, Mapping[str, Any]], Mapping[str, Any]]
     ] = None,
     io_manager_key: Optional[str] = None,
-) -> OpDefinition:
-    code_block = textwrap.indent(node_info["raw_sql"], "    ")
-    description_sections = [
-        node_info["description"],
-        "#### Columns:\n" + _columns_to_markdown(node_info["columns"])
-        if len(node_info["columns"]) > 0
-        else None,
-        f"#### Raw SQL:\n```\n{code_block}\n```",
-    ]
-    description = "\n\n".join(filter(None, description_sections))
-
-    @asset(
-        name=node_info["name"],
-        description=description,
-        non_argument_deps={
-            AssetKey(dep_name.split(".")[-1]) for dep_name in node_info["depends_on"]["nodes"]
-        },
+) -> AssetsDefinition:
+    outs = {}
+    sources = set()
+    out_name_to_node_info = {}
+    internal_asset_deps = {}
+    for node_info in node_infos:
+        node_name = node_info["name"]
+        asset_deps = set()
+        for dep in node_info["depends_on"]["nodes"]:
+            dep_type = dep.split(".")[0]
+            dep_name = dep.split(".")[-1]
+            if dep_type == "source":
+                sources.add(AssetKey(dep_name))
+            asset_deps.add(AssetKey(dep_name))
+        code_block = textwrap.indent(node_info["raw_sql"], "    ")
+        description_sections = [
+            node_info["description"],
+            "#### Columns:\n" + _columns_to_markdown(node_info["columns"])
+            if len(node_info["columns"]) > 0
+            else None,
+            f"#### Raw SQL:\n```\n{code_block}\n```",
+        ]
+        description = "\n\n".join(filter(None, description_sections))
+
+        outs[node_name] = Out(
+            dagster_type=None,
+            asset_key=AssetKey(node_name),
+            description=description,
+            io_manager_key=io_manager_key,
+        )
+        out_name_to_node_info[node_name] = node_info
+        internal_asset_deps[node_name] = asset_deps
+
+    @multi_asset(
+        name="dbt_project",
+        non_argument_deps=sources,
+        outs=outs,
         required_resource_keys={"dbt"},
-        io_manager_key=io_manager_key,
         compute_kind="dbt",
+        internal_asset_deps=internal_asset_deps,
     )
-    def _node_asset(context):
-        context.resources.dbt.run(models=[".".join([node_info["package_name"], node_info["name"]])])
-        if runtime_metadata_fn:
-            metadata = runtime_metadata_fn(context, node_info)
-            yield Output(None, metadata=metadata)
-        else:
-            yield Output(None)
-
-    return _node_asset
+    def _dbt_project_multi_assset(context):
+        dbt_output = context.resources.dbt.run()
```

Comment:
Hi @OwenKephart , following up on https://github.com/dagster-io/dagster/pull/6309#discussion_r802945954 
To make it possible to keep the `models` parameter here so the dbt graph can be split into multiple parts? Use case is still the one highlighted in the comment linked: being able to interleave dbt and python computation (e.g. a machine learning job that runs in between table transformations)

Reply:
hi @amarrella ! the load_assets_from_dbt_project() function accepts the "select" argument, which will allow you to filter for subsets of models (as far as I know, --select is basically equivalent to --models in newer versions of dbt).

However, on closer inspection, this argument doesn't quite work as expected at the moment, but it is a feature that we'll have working for the 0.14.0 release.